### PR TITLE
Add Webhook support

### PR DESCRIPTION
**Pull request by dlitvakb**
Merged (by mistake) and Reverted #55 

Messages from that PR:

> This could be improved if middleman/middleman#1624 is implemented
> Middleman issue being addressed by middleman/middleman#1678

### CODE REVIEW

**Comment by neonichu** [comment]
There's no delay right now between receiving the webhook and rebuilding, which means that it probably won't work as expected a lot of times.

Something also seems to be up with Travis?

**Comment by dlitvakb** [comment]
Re Travis: There is a package failing to download from rubygems, `tins`, locally I can install it just fine, so no idea what's going on.

Re Timeout: I'll add that! What time do you consider to be prudential? (I could also poll the API, but I don't think that's the best way to do it, I'd rather just sleep `x` amount of seconds)

**Comment by neonichu** [comment]
I think we recommend 5 minutes as a delay for them, but probably best to make it configurable, with that value as the default.

**Comment by asaaki** [comment]
@dlitvakb 

`Gem::InstallError: tins requires Ruby version >= 2.0.`

So this won't succeed for Ruby 1.9 builds.

**Comment by dlitvakb** [comment]
Ahhhh! Thanks @asaaki,

Thanks makes a lot of sense.

@pxlpnk Do you think it's ok to remove 1.9.x from TravisCI? They are very outdated ruby versions

**Comment by pxlpnk** [comment]
Ruby 1.9.x is not officially supported anymore. I don't really know how many people still use it.
However `tins` comes from `coveralls` and I can t see this being used or even active.

**Comment by dlitvakb** [comment]
Per discussion with @pxlpnk I'm moving the `WebhookServer` code to a standalone repository: https://github.com/contentful/contentful-webhook-listener.rb

So that it can be extended and reused for other projects

**Comment by dlitvakb** [comment]
@neonichu @pxlpnk Split has been done and code updated accordingly

**Comment by dlitvakb** [comment]
https://github.com/middleman/middleman/pull/1681 Adds `before_server`.

This can now be merged, but can make a follow up release using that hook when it's released.

File: README.md [hunk 1]
```diff
@@ -44,14 +44,16 @@ activate :contentful do |f|
...
+content_types       | Hash describing the mapping applied to entries of the imported content types
+use_preview_api     | Boolean to toggle the used API. Set it to `false` to use `cdn.contentful.com` (default value). Set it to `true` to use `preview.contentful.com`. More info in [the documentation](https://www.contentful.com/developers/documentation/content-delivery-api/#preview-api)
+all_entries         | Boolean to toggle multiple requests to the API for getting over 1000 entries. This uses a naive approach and can get rate limited.
+rebuild_on_webhook  | Boolean to toggle Webhook server. Server will run in port 5678, and will be expecting to receive Contentful Webhook calls on `/receive`
+webhook_timeout     | Integer (in seconds) for wait time after Webhook received for rebuilding. Only used if `:rebuild_on_webhook` is true. Defaults to 5 minutes
```
Comment (neonichu) [inline README.md]: I would change "5 minutes" to "300 seconds" to avoid any possible unit confusions

**Comment by dlitvakb** [comment]
@neonichu fixed

**Pull request merged**